awx/main/models/unified_jobs.py

```diff
@@ -842,15 +842,16 @@ def save(self, *args, **kwargs):
             if 'finished' not in update_fields:
                 update_fields.append('finished')
 
+        if self.elapsed is None:
+            self.elapsed = 0.0
+            if 'elapsed' not in update_fields:
+                update_fields.append('elapsed')
+
         # If we have a start and finished time, and haven't already calculated
         # out the time that elapsed, do so.
-        if self.started and self.finished and not self.elapsed:
+        if self.started and self.finished and self.elapsed == 0.0:
             td = self.finished - self.started
-            elapsed = (td.microseconds + (td.seconds + td.days * 24 * 3600) * 10 ** 6) / (10 ** 6 * 1.0)
-        else:
```

Comment:
this line is the culprit

the issue is that on a job run we call `.save()` multiple times

when `started` and `finished` is defined and elapsed is 0.0, we set it (e.g. 5 seconds) and correctly save this field to the database

now on the very next `.save()`, the `not self.elapsed` condition fails, and we set elapsed back to 0.0

the following conditional is now true, `self.elapsed` (which is 5 seconds) is not equal to 0.0, so we save 0.0 to the database

Reply:
And I suppose an additional call to .save() would set it back to 5 seconds. So it would toggle between 0.0 and 5 seconds on each save.

Reply:
@chrismeyersfsu do you think this is related to the job parentage perf optimizations you made recently?  I can't think of any other change that would explain a side effect like this.